### ISSUE TITLE
Feature extraction

### ISSUE THREAD

**Issue by jjjjjjiayx** [mahmoodlab/CLAM#57]
Hello there,
In order to better understand your code, I carefully read the published article. As a beginner in deep learning, I have encountered some confusion, and I would like to ask you for advice.
The article mentioned "Namely, we take a ResNet50 model pre-trained on ImageNet and use adaptive mean-spatial pooling after the third residual block of the network to convert each 256 × 256 patch into a 1,024-dimensional feature vector using a batch size of 128 per GPU across multiple GPUs."
Why not directly use the existing data set and labels to train ResNet50 and extract features? Or what is the purpose of your training with ImageNet, feature interpretability?
Thank you if you can help.

**Comment by fedshyvana** [comment]
Hi, yes you can train ResNet50 or any CNN of your choice if you have pairs of (image patch, patch label) for your dataset or a subset of your data. The problem we're concerned with here is if you don't have such data but rather, you only have labels for entire slide images, which can each be made of tens of thousands of patches, i.e. the dataset you have is (N patches in a slide, slide label) where N is the number of patches in each slide, and can vary between different slides.
